client/my-sites/site-settings/settings-security/main.jsx

```diff
@@ -85,9 +91,14 @@ SiteSettingsSecurity.propTypes = {
 export default connect( state => {
 	const site = getSelectedSite( state );
 	const siteId = getSelectedSiteId( state );
+	const rewindState = getRewindState( state, siteId ).state;
+	const showRewindCredentials =
```

Comment:
Would it make sense to have a function that is available as a selector. 
So that we don't have to repeat this logic in multiple places?

Reply:
it's a good question and one I struggle with. in cases like `rewindState` where we have a strong data type definition (both in the schema and in design documents) I'm generally more willing to avoid the selector for sub values because the only value the selector brings is type safety. it's easy enough to find all places `rewindState` is being used because we're not passing it around through multiple levels of React components.

right now the absence of type-safety (strong static typing) has meant a proliferation of selectors intended to provide that safety but that proliferation of selectors has introduced needless bloat into the app that our customers pay for 😢 

ideally we'd start using something like TypeScript to provide static typing and eliminate 90% of our selectors while preserving the type safety

Reply:
Thanks for the explanation. 👍